packages/react-instantsearch-core/src/connectors/connectQueryRules.ts

```diff
@@ -0,0 +1,40 @@
+import createConnector from '../core/createConnector';
+import { getResults } from '../core/indexUtils';
+
+export type CustomUserData = {
+  [key: string]: any;
+};
+
+type QueryRulesProps<TItem = CustomUserData> = {
+  transformItems: (items: TItem[]) => TItem[];
+};
+
+export default createConnector({
+  displayName: 'AlgoliaQueryRules',
+
+  defaultProps: {
+    transformItems: items => items,
+  } as QueryRulesProps,
+
+  getProvidedProps(props: QueryRulesProps, _1: any, searchResults: any) {
+    const results = getResults(searchResults, this.context);
+
+    if (results === null) {
+      return {
+        items: [],
+        canRefine: false,
+      };
+    }
+
+    const { userData = [] } = results;
+    const { transformItems } = props;
+    const transformedItems = transformItems
```

Comment:
We have a default value, we don't have to make the assertions. You can fake the default props inside the test.